src/dowel/csv_output.py

```diff
@@ -41,13 +41,33 @@ def record(self, data, prefix=''):
                     extrasaction='ignore')
                 self._writer.writeheader()
 
+            # new fieldnames detected, need to read in current file, modify, and write everything again
             if to_csv.keys() != self._fieldnames:
-                self._warn('Inconsistent TabularInput keys detected. '
```

Comment:
Since you deleted this code, is there any other code in this file that is no longer needed and can also be deleted? :)